src/Vendr.PaymentProviders.Opayo/Api/OpayoClient.cs

```diff
@@ -61,41 +61,12 @@ public CallbackResult HandleCallback(OrderReadOnly order, CallbackRequestModel r
 
         private string GetMethodUrl(string type, bool testMode)
         {
-            switch (testMode)
+            if (testMode)
             {
-                case false:
-                    switch (type.ToUpperInvariant())
-                    {
-                        case "AUTHORISE":
-                            return "https://live.sagepay.com/gateway/service/authorise.vsp";
-                        case "PAYMENT":
-                        case "DEFERRED":
-                        case "AUTHENTICATE":
-                            return "https://live.sagepay.com/gateway/service/vspserver-register.vsp";
-                        case "CANCEL":
-                            return "https://live.sagepay.com/gateway/service/cancel.vsp";
-                        case "REFUND":
-                            return "https://live.sagepay.com/gateway/service/refund.vsp";
-                    }
-                    break;
-                case true:
-                    switch (type.ToUpperInvariant())
-                    {
-                        case "AUTHORISE":
-                            return "https://test.sagepay.com/gateway/service/authorise.vsp";
-                        case "PAYMENT":
-                        case "DEFERRED":
-                        case "AUTHENTICATE":
-                            return "https://test.sagepay.com/gateway/service/vspserver-register.vsp";
-                        case "CANCEL":
-                            return "https://test.sagepay.com/gateway/service/cancel.vsp";
-                        case "REFUND":
-                            return "https://test.sagepay.com/gateway/service/refund.vsp";
-                    }
-                    break;
+                return OpayoEndpoints.TestEndpoints[type.ToUpperInvariant()];
             }
 
-            return string.Empty;
+            return OpayoEndpoints.LiveEndpoints[type.ToUpperInvariant()];
```

Comment:
Same thing I mentioned on the Umbraco Commerce PR. Lets do a pre-check the key exists in the dictionary first and throw a more meaningful error if it doesn't exist.

Reply:
I refactored to throw a new exception when the enpoint type does not exist.